src/screens/SignIn/index.tsx

```diff
@@ -1,13 +1,65 @@
-import React, { FC } from 'react';
-import fire from '../../services/firebase';
-
-const SignIn: FC = () => (
-  <div
-    onClick={() => {
-      console.log(fire);
-    }}>
-    Sign In
-  </div>
-);
+import React, { FC, useEffect } from 'react';
+import { fireApi } from '../../services/firebase';
+import { routes } from '../../router';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { userActions, userOperations, userSelectors } from '../../redux/user';
+import { RootState } from '../../redux/rootReducer';
+import { Redirect } from 'react-router-dom';
+import {
+  FacebookLoginButton,
+  GoogleLoginButton,
+} from 'react-social-login-buttons';
+import EmailAndPasswordSignIn from './EmailAndPasswordSignIn';
+
+const SignIn: FC = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(userOperations.checkSocialMediaRedirect());
+  }, [dispatch]);
+
+  const googleSignIn = () => {
+    dispatch(userActions.loginRequest());
```

Comment:
i think there is no point in dispatching loginRequest here, because we get redirected and loose the state anyway